src/couch_scanner/src/couch_scanner_plugin_conflict_finder.erl

```diff
@@ -0,0 +1,198 @@
+    DocReport = maps:merge(ConflictsReport, DeletedConflictsReport),
+    report_per_doc(#st{} = St, DbName, DocId, DocReport),
+    DbReport = maps:map(fun(_K, V) when is_list(V) -> length(V) end, DocReport),
+    report(#st{} = St, DbName, DbReport);
+check(#st{opts = Opts} = St, DbName, _DocId, Revs) ->
+    {DeletedConflicts, Conflicts} =
+        lists:partition(fun(R) -> R#rev_info.deleted end, Revs),
+    ConflictsReport = gen_report(db, ?CONFLICTS, Opts, Conflicts),
+    DeletedConflictsReport = gen_report(db, ?DELETED_CONFLICTS, Opts, DeletedConflicts),
+    DbReport = maps:merge(ConflictsReport, DeletedConflictsReport),
+    report(#st{} = St, DbName, DbReport).
+
+gen_report(db, ?CONFLICTS, #{?CONFLICTS := true}, Revs) ->
+    #{?CONFLICTS => length(Revs)};
+gen_report(db, ?DELETED_CONFLICTS, #{?DELETED_CONFLICTS := true}, Revs) ->
+    #{?DELETED_CONFLICTS => length(Revs)};
+gen_report(doc, ?CONFLICTS, #{?CONFLICTS := true}, Revs) ->
+    #{?CONFLICTS => [couch_doc:rev_to_str(R#rev_info.rev) || R <- Revs]};
+gen_report(doc, ?DELETED_CONFLICTS, #{?DELETED_CONFLICTS := true}, Revs) ->
+    #{?DELETED_CONFLICTS => [couch_doc:rev_to_str(R#rev_info.rev) || R <- Revs]};
+gen_report(_Type, _Key, #{} = _Opts, _Revs) ->
+    #{}.
+
+report(#st{} = St, DbName, Report) ->
+    #st{report = Total, dbname = PrevDbName} = St,
+    case is_binary(PrevDbName) andalso DbName =/= PrevDbName of
+        true ->
+            % We switched dbs, so report stats for old db
+            % and make the new one the current one
+            report_per_db(St, PrevDbName, Total),
+            St#st{report = Report, dbname = DbName};
+        false ->
+            % Keep accumulating per-db stats
+            St#st{report = merge_report(Total, Report), dbname = DbName}
+    end.
+
+merge_report(#{} = Total, #{} = Update) ->
+    Fun = fun(_K, V1, V2) -> V1 + V2 end,
+    maps:merge_with(Fun, Total, Update).
+
+report_per_db(#st{sid = SId}, DbName, #{} = Report) when
+    map_size(Report) > 0, is_binary(DbName)
+->
+    {Fmt, Args} = report_fmt(Report),
+    Meta = #{sid => SId, db => DbName},
+    ?WARN(Fmt, Args, Meta);
+report_per_db(#st{}, _, _) ->
+    ok.
+
+report_per_doc(#st{sid = SId}, DbName, DocId, Report) when
+    map_size(Report) > 0, is_binary(DbName)
+->
+    {Fmt, Args} = report_fmt(Report),
+    Meta = #{sid => SId, db => DbName, doc => DocId},
+    ?WARN(Fmt, Args, Meta);
+report_per_doc(#st{}, _, _, _) ->
+    ok.
+
+report_fmt(Report) ->
+    Sorted = lists:sort(maps:to_list(Report)),
+    FmtArgs = [{"~s:~p ", [K, V]} || {K, V} <- Sorted],
+    {Fmt1, Args1} = lists:unzip(FmtArgs),
+    Fmt2 = lists:flatten(Fmt1),
+    Args2 = flatten_one_level(Args1),
+    {Fmt2, Args2}.
+
+flatten_one_level(List) when is_list(List) ->
+    case lists:flatten(List) =:= List of
+        true ->
+            List;
+        false ->
+            lists:append([
+                case is_list(E) of
```

Comment:
I think here we're handling the case of list of binaries `[<<"2-y">>,<<"2-x">>]` vs a number say `2`?

Perhaps we could shorten a bit the string by translating the binaries to string so it would look like '["2-y", "2-x"]'?

Another potential issues is if we have a large amount of revs, say a 1000 (we've seen documents like that), in that case often the log line gets cut, not sure what could do there, maybe nothing? We could emit it as a number then "we found conflicts but it's a 1000 so we'll just say 1000 instead of [..., ..., ... x 1000]?

Reply:
Thank you for the suggestion!
Convert them back to strings and add a `max_revs` option to control how many revs are shown in the doc_report (default = 10). Also add a comment in `default.ini`, if we want to get the full list, we can use `GET /{db}/{docid}?conflicts=true` endpoint.